app/controllers/doctors_controller.rb

```diff
@@ -0,0 +1,64 @@
+class DoctorsController < ApplicationController
+  before_action :set_doctor, only: %i[show update destroy]
+
+  # GET /doctors
+  def index
+    @doctors = Doctor.all
+    i = 0
+    @docs = []
+    while i < @doctors.length
+      doc = {}
+      doc['doctor'] = @doctors[i]
+      doc['address'] = Address.find_by(doctor_addresses: DoctorAddress.find_by(doctor: @doctors[i]))
+      @docs.push(doc)
+      i += 1
```

Comment:
- [ ] I suggest that you try doing this in a better way, maybe creating a method in the `doctor` model could be helpful, because personally I don't think it's a good practice to do all of this here in the controller.